documentation/svfsiplus/theory/fluid/readme.md

```diff
@@ -0,0 +1,258 @@
+The Cauchy stress tensor is $\boldsymbol{\sigma} = \boldsymbol{\sigma}\left(\boldsymbol{x}, t\right) = -p\boldsymbol{I} + 2\mu\left(\boldsymbol{u}\right)\epsilon$, where  $\epsilon = \epsilon\left(\boldsymbol{u}\right) = \nabla^{s} \boldsymbol{u} = \frac{1}{2}\left(\nabla \boldsymbol{u} + \left(\nabla\boldsymbol{u}\right)^{\text{T}} \right)$ is the strain rate tensor. The effective dynamic viscosity, $\mu\left(\boldsymbol{u}\right)$, is written generally as a function of velocity here to account for non-Newtonian fluids. For Newtonian fluids, $\mu$ is a simply constant. The divergence of the Cauchy stress tensor, written in both vector and index notation, is
+
+$$
+\boldsymbol{\nabla} \cdot \boldsymbol{\sigma} = -\boldsymbol{\nabla}p + 2\boldsymbol{\epsilon}\boldsymbol{\nabla}\mu + \mu\nabla^{2}\boldsymbol{u},
+$$
+
+$$
+\sigma_{ij,j} = -p_{,i} + 2\epsilon_{ij}\frac{\partial \mu}{\partial x_{j}} + \mu\nabla^{2}u_{i},
+$$
+
+The boundary conditions are 
+
+$$
+\boldsymbol{u} = \boldsymbol{g},
+$$
+
+$$
+\boldsymbol{\sigma} \cdot \boldsymbol{n} = \boldsymbol{h},
+$$
+
+where $\boldsymbol{g}$ is the prescribed velocity and $\boldsymbol{h}$ is the prescribed traction.
+
+We will solve the Navier-Stokes and Navier-Stokes-Brinkman equations numerically, using the finite element method for spatial discretization <a href="#ref-8">[8]</a>.
+
+### Standard (Galerkin) weak form
+    
+For the finite element method, we will first derive the Galerkin weak form for the Navier-Stokes-Brinkman equations. We define our trial and weighting function spaces,
+
+$$
+u_{i} \in \tau_{i} = \{u_{i} \in H^{1}\left(\Omega\right) \mid u_{i} = g_{i} \text{ on } \Gamma_{g_{i}}\},
+$$
+
+$$
+w_{i} \in \nu_{i} = \{w_{i} \in H^{1}\left(\Omega\right) \mid w_{i} = 0 \text{ on } \Gamma_{g_{i}}\},
+$$
+
+$$
+p, q \in Q = \{p \in L^{2}\left(\Omega\right)\},
+$$
+
+where $\boldsymbol{w}$ is the weighting function for velocity and $q$ is the weighting function for pressure. We represent these weighting functions discretely on a per-element-basis as
+
+$$
+w_{i} = \sum_{a=1}^{n_{en}} N_{a}^{w}w_{ai},
+$$
+
+$$
+q = \sum_{a=1}^{n_{en}} N_{a}^{q}q_{a},
+$$
+
+where $N_{a}^{w}$ and $N_{a}^{q}$ are the nodal shape (basis) functions for the velocity and pressure spaces, respectively, and $w_{ai}$ and $q_{a}$ are the associated arbitrary nodal coefficients. Similarly, the trial functions are represented by
+
+$$
+u_{i} = \sum_{a=1}^{n_{en}} N_{a}^{w}u_{ai},
+$$
+
+$$
+p = \sum_{a=1}^{n_{en}} N_{a}^{q}p_{a}.
+$$
+
+We then multiply the Navier-Stokes-Brinkman equations by $\boldsymbol{w}$ and $q$, respectively, and integrate by parts to obtain the standard Galerkin momentum and continuity weak forms <a href="#ref-7">[7]</a> <a href="#ref-5">[5]</a>,
+
+$$
+\int \rho w_{i}\frac{du_{i}}{dt} \,d\Omega + \int \rho w_{i}u_{k}u_{i, k} \,d\Omega + \int w_{i, j}\sigma_{ij} \,d\Omega + \int \frac{\mu}{K}w_{i}u_{i} \,d\Omega - \int w_{i}\rho b_{i} \,d\Omega - \int_{\Gamma_{h}} w_{i}h_{i} \,d\Gamma = 0,
```

Comment:
@JonathanPham I would add _\Omega for each volumetric integral for clarity.

Reply:
Good idea.